scipy/signal/signaltools.py

```diff
@@ -189,7 +190,28 @@ def correlate(in1, in2, mode='full', method='auto'):
     >>> ax_corr.set_title('Cross-correlated with rectangular pulse')
     >>> ax_orig.margins(0, 0.1)
     >>> fig.tight_layout()
-    >>> fig.show()
+    >>> plt.show()
+
+    Compute the cross-correlation of a noisy signal with the original signal.
+
+    >>> x = np.arange(128) / 128
+    >>> sig = np.sin(2 * np.pi * x)
+    >>> sig_noise = sig + np.random.randn(len(sig))
+    >>> corr = signal.correlate(sig_noise, sig)
+    >>> corr /= np.max(corr)
+
+    >>> fig, (ax_orig, ax_noise, ax_corr) = plt.subplots(3, 1)
+    >>> ax_orig.plot(sig)
+    >>> ax_orig.set_title('Original signal')
+    >>> ax_noise.plot(sig_noise)
+    >>> ax_noise.set_title('Signal with noise')
+    >>> ax_corr.plot(np.arange(-127, 128), corr)
+    >>> ax_corr.set_title('Cross-correlated signal')
+    >>> ax_orig.margins(0, 0.1)
+    >>> ax_noise.margins(0, 0.1)
+    >>> ax_corr.margins(0, 0.1)
```

Comment:
Axis labels are always helpful. In this case at least labeling the x axes for the signals with `Sample number` and the cross-correlated one with `Lag` would help. Also, you can make use of correlation_lags rathor than manually computing the `np.arange`.

Reply:
Fixed.